---
title: "Classification of Neisseria meningitidis genomes with a bag-of-words approach and machine learning"
collection: publications
permalink: /publications/2024-Classification-of-Neisseria-meningitidis-genomes-with-a-bag-of-words-approach-and-machine-learning
date: 2024-02-13
venue: 'iScience, Elsevier'
---

Recommended citation:
Marco Podda, Simone Bonechi, Andrea Palladino, Mattia Scaramuzzino, Alessandro Brozzi, Guglielmo Roma, ... & Margherita Bodini. Classification of Neisseria meningitidis genomes with a bag-of-words approach and machine learning. iScience. 2024. ([BibTex](http://clem.diism.unisi.it/~coco_ts/S2589004224004784.bib))
{: style="text-align: justify"}

### Abstract
Whole genome sequencing of bacteria is important to enable strain classification. Using entire genomes as an input to machine learning (ML) models would allow rapid classification of strains while using information from multiple genetic elements. We developed a “bag-of-words” approach to encode, using SentencePiece or k-mer tokenization, entire bacterial genomes and analyze these with ML. Initial model selection identified SentencePiece with 8,000 and 32,000 words as the best approach for genome tokenization. We then classified in Neisseria meningitidis genomes the capsule B group genotype with 99.6% accuracy and the multifactor invasive phenotype with 90.2% accuracy, in an independent test set. Subsequently, in silico knockouts of 2,808 genes confirmed that the ML model predictions aligned with our current understanding of the underlying biology. To our knowledge, this is the first ML method using entire bacterial genomes to classify strains and identify genes considered relevant by the classifier.
{: style="text-align: justify"}

You can find the full paper [here](https://linkinghub.elsevier.com/retrieve/pii/S2589004224004784)
{: style="text-align: justify"}
